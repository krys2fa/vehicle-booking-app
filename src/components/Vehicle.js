import React from 'react';
import PropTypes from 'prop-types';

const Vehicle = ({ vehicle }) => {
  const {
    name, photo,
  } = vehicle;

  return (
    <div className="card width">
      <img
        className="card-img-top"
        src={photo}
        alt=""
      />
      <div className="card-body">
        <h4 className="card-title">{name}</h4>
        <button type="button" className="btn btn-success">
          See Details
        </button>
      </div>
    </div>
  );
};

export default Vehicle;

Vehicle.propTypes = {
  vehicle: PropTypes.shape({
    name: PropTypes.string,
    photo: PropTypes.string,
  }).isRequired,
};
