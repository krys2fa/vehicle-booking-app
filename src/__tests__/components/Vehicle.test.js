import React from 'react';
import { render } from '@testing-library/react';
import Vehicle from '../../components/Vehicle';

it('renders the Vehicle component correctly', () => {
  const vehicle = {
    name: 'Daewoo Lemans',
    transmission: 'Manual',
    fee: '1500',
  };
  const { asFragment } = render(<Vehicle vehicle={vehicle} />);
  expect(asFragment()).toMatchSnapshot();
});
