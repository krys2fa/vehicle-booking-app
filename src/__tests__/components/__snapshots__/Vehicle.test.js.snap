// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Vehicle component correctly 1`] = `
<DocumentFragment>
  <div
    class="card width"
  >
    <img
      alt=""
      class="card-img-top"
    />
    <div
      class="card-body"
    >
      <h4
        class="card-title"
      >
        Daewoo Lemans
      </h4>
      <button
        class="btn btn-success"
        type="button"
      >
        See Details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
